### PR TITLE
Fix wrong tags generated by Oracle custom queries

### PR DESCRIPTION
**Pull request by nenadnoveljic**
### What does this PR do?

This PR fixes the issue with wrong tags in Oracle custom queries. It also adds a test case for detecting the problem in the future.

### Additional Notes

The bug was caused by incorrectly copying slices.

https://datadoghq.atlassian.net/browse/DBMON-1995

### Describe how to test/QA your changes

The configuration for reproducing the problem:

```
    tags:  # Optional
      - tag1:vtag1
      - tag2:vtag2
    custom_queries:
      - metric_prefix: oracle.custom_query.vw_web_extraction_monitor
        pdb: XEPDB1
        query: |
          select selection_id, user_schema
          from DMR_WEB.VW_WEB_EXTRACTION_MONITOR
          where selection_id is not NULL order by selection_id
        columns:
          - name: selection_id
            type: gauge
          - name: user_schema
            type: tag
```
Check that every metric doesn't have the tag of the last row.

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: pkg/collector/corechecks/oracle-dbm/custom_queries.go [hunk 1]
```diff
@@ -120,27 +128,34 @@ func (c *Check) CustomQueries() error {
 					if v_str, ok := v.(string); ok {
 						metricRow.value, err = strconv.ParseFloat(v_str, 64)
 						if err != nil {
-							logTypeError(metricPrefix, "number", metricRow.name, v, q.Query)
+							logTypeError(metricPrefix, "number", metricRow.name, v, q.Query, err)
```
Comment (lu-zhengda) [inline pkg/collector/corechecks/oracle-dbm/custom_queries.go:131]: Why only log this typing error but not add to `allErrors`?

**Review by estherk15 — APPROVED** [review]

File: pkg/collector/corechecks/oracle-dbm/custom_queries.go (hunk 1 above)
Comment (nenadnoveljic) [reply pkg/collector/corechecks/oracle-dbm/custom_queries.go]: Fixed now, thx.

File: pkg/collector/corechecks/oracle-dbm/custom_queries.go [hunk 2]
```diff
@@ -65,15 +69,16 @@ func (c *Check) CustomQueries() error {
 		"histogram":       sender.Histogram,
 		"historate":       sender.Historate,
 	}
+	var allErrors error
```
Comment (alexandre-normand) [inline pkg/collector/corechecks/oracle-dbm/custom_queries.go:72]: I think the joining of multiple errors might be a good use case for the new (as of go 1.20) [errors.Join](https://pkg.go.dev/errors#Join).

**Review by lu-zhengda — APPROVED** [review]

File: pkg/collector/corechecks/oracle-dbm/custom_queries.go (hunk 2 above)
Comment (nenadnoveljic) [reply pkg/collector/corechecks/oracle-dbm/custom_queries.go:72]: That's what I implemented in the first place, but the next custom Oracle build has to be made on top of `7.47.0`, which has to be compiled with an earlier `Go` version. We can change this, once `7.48` is out.

Comment (alexandre-normand) [reply pkg/collector/corechecks/oracle-dbm/custom_queries.go:72]: ah, makes sense. 💯

**Pull request merged**